wrappers/services.go

```diff
@@ -141,6 +171,20 @@ func AddSnapServices(s *snap.Info, inter interacter) (err error) {
 			return err
 		}
 		written = append(written, svcFilePath)
+
+		// Generate systemd .socket files if needed
+		socketFiles, err := generateSnapSocketFiles(app)
+		if err != nil {
+			return err
+		}
+		for path, content := range *socketFiles {
+			os.MkdirAll(filepath.Dir(path), 0755)
```

Comment:
Isn't the path generated by systemd?

Reply:
are you talking about the `.socket` file or the directory contaning it?
We probably don't need the `os.MkdirAll` indeed, since `/etc/systemd/system` should always exist

Reply:
and if it doesn't exist, `/etc/systemd` is readonly on core